flox/core.py

```diff
@@ -468,8 +468,26 @@ def factorize_(
                 bins = np.concatenate([expect.left.to_numpy(), [expect.right[-1].to_numpy()]])
             else:
                 bins = np.concatenate([expect.left.to_numpy(), [expect.right[-1]]])
+
             # code is -1 for values outside the bounds of all intervals
-            idx = pd.cut(flat, bins=bins, right=expect.closed_right).codes.copy()
+            # idx = pd.cut(flat, bins=bins, right=expect.closed_right).codes.copy()
+
+            # digitize is 0 or idx.max() for values outside the bounds of all intervals
+            # make it behave like pd.cut:
+            if len(bins) > 1:
+                right = expect.closed_right
+                idx = np.digitize(
+                    flat,
+                    bins=bins.view(np.intp) if bins.dtype.kind == "M" else bins,
+                    right=right,
+                )
+                idx = pd.to_numeric(idx, downcast="integer")
```

Comment:
`pd.to_numeric(idx, downcast="integer")` is used to downcast and match the tests dtype.
But `to_numeric` is slow. There's no performance gain compared to `pd.cut` if we have to match the dtype.

Reply:
What tests are these? Presumably they test factorize. I don't think we need to match dtype. Is it just that they are always `int64` now instead of `intp`?

Reply:
Without downcasting idx became int64, but pd.cut was returning int8. 

I'll comment it out and we'll see which tests fail now.